extensions/logging-gelf/runtime/src/main/java/io/quarkus/logging/gelf/AdditionalFieldConfig.java

```diff
@@ -0,0 +1,26 @@
+package io.quarkus.logging.gelf;
+
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigGroup;
+import io.quarkus.runtime.annotations.ConfigItem;
+
+/**
+ * Post additional fields. E.g. `fieldName1=value1,fieldName2=value2`.
+ */
+@ConfigGroup
+public class AdditionalFieldConfig {
+    /**
+     * Additional field value.
+     */
+    @ConfigItem
+    public String value;
+
+    /**
+     * Additional field type specification .
+     * Supported types: String, long, Long, double, Double and discover.
+     * Discover is the default if not specified, it discovers field type based on parseability.
+     */
+    @ConfigItem
+    public Optional<String> type;
```

Comment:
Maybe make it a `String` and make `discover` the `defaultValue`? This way, it's all auto-documented?

I was wondering if we could use an enum for that but I'm pretty sure the magic case handling we added will conflict with having `Long` and `long`.

Reply:
I'm not comfortable with enum because of `Long` and `long`, or we can choose to only support one of them as both will ends up the same I think ...

If I have no value, I didn't emit any type configuration at all. If I add a default value the default type configuration will be created for each additional type. I have no string opinion on this ... as you prefere ...

Reply:
I think I prefer having a `defaultValue`.

Reply:
Done